### ISSUE TITLE
Echidna unable to detect constants in storage Structs

### ISSUE THREAD

**Issue by jat9292** [crytic/echidna#1014]
### Echidna should be able to detect constants in Structs

Hello.
I just noticed that echidna is unable to chose values which are defined as constants inside structs in the solidity code. 
Actually I stumbled on this issue while trying to build my own fuzzer, using the same technique to mine constants that echidna is using. I was unsure if I had to post the issue here or in the Slither repo, as in order to solve it we must improve the `_extract_constants` function from the [Slither](https://github.com/crytic/slither/blob/master/slither/printers/guidance/echidna.py#L225) package to make it able to detect such constants.

Here is a minimal example which current version of echidna fails to solve:

```solidity
contract Test {
    struct Todo {
        uint x;
        bool completed;
    }
    bool flag = true;
    Todo todo = Todo(129856743768, false);

    function set(uint a) public {
        if (a == todo.x) flag = false;
    }

    function echidna_test_balance() public view returns (bool) {
        return flag;
    }
}
```
Maybe if I have time I will try to do a pull request on the Slither repo, unless I am missing something and this could work with a better config?

EDIT : sorry I think the issue is not what I thought : indeed , weirdly this version is easily broken by echidna : 
```solidity
contract Test {
    struct Todo {
        uint x;
        bool completed;
    }
    bool flag = true;

    function set(uint a) public {
        Todo memory todo = Todo(129856743768, false);
        if (a == todo.x) flag = false;
    }

    function echidna_test_balance() public view returns (bool) {
        return flag;
    }
}
```
So the problem does not seem to be related to the constant inside the struct, but to the fact that it is declared as a state variable instead of a local one, this behavior is still a bit weird in my opinion...

**Comment by montyly** [comment]
Thanks @jat9292 , that's a great catch. We will investigate it (https://github.com/crytic/slither/issues/1828)

**Comment by ggrieco-tob** [comment]
This is still an issue, and indeed slither fails to get the constant needed

**Comment by smonicas** [comment]
This was fixed with slither 0.10.0
